modules/apps/translation/translation-web/src/main/java/com/liferay/translation/web/internal/portlet/action/UpdateTranslationMVCActionCommand.java

```diff
@@ -294,6 +295,25 @@ private String _getRedirect(
 		).setParameter(
 			"backURLTitle", ParamUtil.getString(actionRequest, "backURLTitle")
 		).buildString();
+
+		String namespace = _portal.getPortletNamespace(themeDisplay.getPpid());
+
+		for (Map.Entry<String, String[]> entry :
+				_getInfoFieldParameterValues(
+					actionRequest
+				).entrySet()) {
+
+			String[] values = entry.getValue();
+
+			if (ArrayUtil.isEmpty(values)) {
+				continue;
+			}
+
+			redirect = HttpComponentsUtil.addParameter(
+				redirect, namespace + entry.getKey(), entry.getValue()[0]);
+		}
+
+		return redirect;
```

Comment:
Why convert the url to a string? Doesn't that make adding parameters much more cumbersome? I think it would be better to create the template URL as a `PortletURL`, and then add the necessary parameters via `setParameter`.

Also, be careful with chaining call (e.g. `entrySet' in the foreach statement and `entry.getValue()[...]`).